csclient/csclient.go

```diff
@@ -36,6 +36,11 @@ import (
 	"gopkg.in/juju/charmrepo.v4/csclient/params"
 )
 
+const (
+	userAgentKey   = "User-Agent"
+	userAgentValue = "Golang_CSClient/4.0"
+)
+
```

Comment:
this helps with the cs client library but ideally we'd be able to tell juju from charm tool and the version of each when they hit to help tell about usage of older clients/etc.

Reply:
You can provide an override, this just states if it's not there.

Reply:
@mitechie added testing for both scenarios:

 1. Default header sent
 2. Overriding header